### PR TITLE
Rework overflow code

### PR DESCRIPTION
**Pull request by adamlwgriffiths**
Support edge visibility on the right side.
Unify the left/right overflow code.
Unify the text rendering code.
Add some comments.
Add a clear function.

The +/- 1's are pretty confusing but
it's mainly because of the cursor being
a physical part of the input throwing
a spanner in these calcs.

### CODE REVIEW

File: src/textual_inputs/text_input.py [hunk 1]
```diff
@@ -170,22 +175,37 @@ def on_focus_handler_name(self, handler_name: str) -> None:
         """
         self._on_focus_message_class = make_message_class(handler_name)
 
+    async def on_focus(self, event: events.Focus) -> None:
```
Comment (adamlwgriffiths) [inline src/textual_inputs/text_input.py:178]: I just moved these so the render code is co-located

File: src/textual_inputs/text_input.py [hunk 2]
```diff
@@ -170,22 +175,37 @@ def on_focus_handler_name(self, handler_name: str) -> None:
...
-        else:
-            if len(self.value) == 0:
-                if self.title and not self.placeholder:
-                    segments = [self.title]
-                else:
-                    segments = [self.placeholder]
-            else:
-                segments = [self._modify_text(self.value)]
-
+        segments = self._render_text()
```
Comment (adamlwgriffiths) [inline src/textual_inputs/text_input.py:208]: Rendering text is now unified

File: src/textual_inputs/text_input.py [hunk 3]
```diff
@@ -208,110 +228,109 @@ def render(self) -> RenderableType:
...
-        # remove 2, 1 for each of the border's edges
-        # remove 1 more for the cursor
-        # remove 2 for the padding either side of the input
+    def _view_width(self):
+        """Width in characters of the inside of the input, including the cursor"""
+        # remove 2, for each of the left/right border edges
+        # remove 2, for the padding either side of the input
         width, _ = self.size
         if self.border:
             width -= 2
```
Comment (adamlwgriffiths) [inline src/textual_inputs/text_input.py:263]: The cursor is accounted for as part of the rendering

**Comment by sirfuzzalot** [comment]
Textual now supports inputs natively -> https://textual.textualize.io/widgets/input/

**Pull request closed without merging**